R/nba_scraper.R

```diff
@@ -1,26 +1,154 @@
+  if ((!is.null(nba_data_env_name)) & (!assertthat::is.string(nba_data_env_name))) {
+    stop("'nba_data_env_name' must be a string if it is not a NULL input")
+  }
+
+  # Create url
+  url_season_year <- season_year +1
+  url_season_type <- ifelse(season_type == "regular", 2, 3)
+  url <-paste0("https://www.espn.com/nba/stats/player/_/season/", url_season_year, "/seasontype/", url_season_type)
+
+  # Set up parameters for remoteDriver
+  remDr <- RSelenium::remoteDriver(
+    remoteServerAddr = "localhost",
+    port = port,
+    browserName = sel_browser
+  )
+
+  # Print message
+  print("Scraping commencing. Please wait!")
+
+  # Initiate Driver
+  remDr$open()
+
+  # Use driver to browse the url
+  remDr$navigate(url)
+
+  # Dynamic loading on webpage to load all the data
+  chk <- FALSE
+  while (!chk) {
+    res <- try({
+      suppressMessages(showmore_button <- remDr$findElement(using = "xpath", "//*[@id='fittPageContainer']/div[3]/div[1]/div/section/div/div[3]/div/a"))
+    }, silent = TRUE)
+    if (class(res) == "try-error") {
+      chk <- TRUE
+    } else {
+      showmore_button$clickElement()
+      # Sleep for 2 seconds for the page to load
+      Sys.sleep(2L)
+    }
+  }
+
+  # Compilation List of plyr XML paths
+  ply_trs <- remDr$getPageSource()[[1]] %>%
+    xml2::read_html() %>%
+    rvest::html_nodes(xpath="//*[@id='fittPageContainer']/div[3]/div[1]/div/section/div/div[3]/section/div[2]/table/tbody/tr")
+
+  stats_trs <- remDr$getPageSource()[[1]] %>%
+    xml2::read_html() %>%
+    rvest::html_nodes(xpath="//*[@id='fittPageContainer']/div[3]/div[1]/div/section/div/div[3]/section/div[2]/div/div[2]/table/tbody/tr")
+
+  # Close Selenium driver
+  remDr$close()
+
+  # Initiate compilation df
+  compiled_df <- data.frame(matrix(, nrow=length(ply_trs), ncol=23))
+  colnames(compiled_df) <- c("Name", "Team", "Position",
+                             "GP", "MIN", "PTS", "FGM", "FGA", "FG%", "3PM", "3PA",
+                             "3P%", "FTM", "FTA", "FT%", "REB", "AST", "STL", "BLK",
+                             "TO", "DD2", "TD3", "PER")
+
+
+  for (i in seq(length(ply_trs))) {
+    # From player
+    player <- ply_trs[i]
+    # player: Player name
+    player_name <- player %>%
+      html_nodes("[class='AnchorLink']") %>%
+      html_text()
+    # player: Team name
+    player_team <- player %>%
+      html_nodes("[class='pl2 n10 athleteCell__teamAbbrev']") %>%
+      html_text()
+
+    # From stats
+    stats = stats_trs[i]
+    # stats: list of all stats
+    stats_list <- stats %>%
+      html_nodes("td") %>%
+      html_text()
+
+    # Input into compiled df
+    compiled_df[i,1] <- player_name
+    compiled_df[i,2] <- player_team
+    compiled_df[i,3:23] <- stats_list
+  }
+  # Write to csv
+  write.csv(compiled_df, csv_path)
+
+  # If nba_data_env_name is given, create a tibble in Global Env
+  if (!is.null(nba_data_env_name)) {
+    assign(nba_data_env_name, tibble::tibble(compiled_df), envir = .GlobalEnv)
+  }
```

Comment:
Instead of assinging the object to the environment we should return the dataframe

Reply:
this has been changed accordingly with a more recent commit!